src/context/Store.tsx

```diff
@@ -0,0 +1,74 @@
+import { createContext, useReducer } from 'react';
+import { getLocalStorageData } from '../services/getLocalStorageData';
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+type Action =
+  | { type: 'add'; payload: string }
+  | { type: 'delete'; payload: number }
+  | { type: 'update'; payload: Todo }
+  | { type: 'filterList'; payload: Filter }
+  | { type: 'setAllComplete'; payload: boolean }
+  | { type: 'clearAll' };
+
+function reducer(todos: Todo[], action: Action) {
+  let newTodos: Todo[] = [];
+
+  switch (action.type) {
+    case 'add':
+      newTodos = [
+        ...todos,
+        { id: +new Date(), title: action.payload.trim(), completed: false },
+      ];
+      break;
+
+    case 'delete':
+      newTodos = todos.filter(t => t.id !== action.payload);
+      break;
+
+    case 'update':
+      newTodos = todos.map(t =>
+        t.id === action.payload.id ? action.payload : t,
+      );
+      break;
+
+    case 'filterList':
+      newTodos = todos.map(t => ({ ...t, filter: action.payload }));
+      break;
+
+    case 'setAllComplete':
+      newTodos = todos.map(t => ({ ...t, completed: action.payload }));
+      break;
+
+    case 'clearAll':
+      newTodos = todos.filter(t => !t.completed);
+      break;
+
+    default:
+      return todos;
+  }
```

Comment:
You got the idea.
Here is the next improvement steps:
1. Use object literals to be more readable (I do prefer using it to translation in all cases, but its up to you, totally optional): https://blog.rocketseat.com.br/substituindo-a-instrucao-switch-por-object-literal/
2. You can bring all states related from todo to this reducer instead of using only the returned array, such as input state and isEditing. (optional too, practice it for the next study case)

Reply:
For example
setTextUpdate state should be updated by
1. searching the edited item by its id
2. changing directly the todo state